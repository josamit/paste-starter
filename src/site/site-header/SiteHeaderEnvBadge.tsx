import React from "react";
import { Badge } from "@twilio-paste/core";
import { BadgeVariants } from "@twilio-paste/badge/dist/types";
import { useWindowObject } from "@hooks/useWindowObject";

const determineBadgeVariant = (environment: string) => {
  let variant: BadgeVariants = "error";

  if (environment === "stage") {
    variant = "warning";
  }

  if (environment === "dev") {
    variant = "success";
  }

  return variant;
};

const SiteHeaderEnvBadge: React.FC = () => {
  const { environment } = useWindowObject();

  const variant = determineBadgeVariant(environment);

  return (
    <Badge as="span" variant={variant}>
      {environment}
    </Badge>
  );
};

export default SiteHeaderEnvBadge;
