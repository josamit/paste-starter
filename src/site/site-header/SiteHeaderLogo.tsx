import * as React from "react";
import { Text } from "@twilio-paste/text";
import { Box } from "@twilio-paste/box";
import {
  MediaBody,
  MediaFigure,
  MediaObject,
} from "@twilio-paste/media-object";
import { styled } from "@twilio-paste/styling-library";
import { Link } from "react-router-dom";
import { LogoTwilioIcon } from "@twilio-paste/icons/esm/LogoTwilioIcon";
import { Stack } from "@twilio-paste/stack";
import SiteHeaderEnvBadge from "./SiteHeaderEnvBadge";

const StyledLink = styled(Link)`
  text-decoration: none;
  color: inherit;

  &:visited {
    color: inherit;
  }
`;

interface SiteHeaderLogoProps {
  title?: string;
  subtitle?: string;
}

// Note: 'subtitle' isn't passed for the mobile view, so we use that fact
// to render different sizes and spacing in mobile
const SiteHeaderLogo: React.VFC<SiteHeaderLogoProps> = ({
  title,
  subtitle,
}) => {
  return (
    <Box
      display="flex"
      alignItems="center"
      paddingRight="space70"
      paddingLeft="space70"
      borderRightColor="colorBorderInverseWeaker"
      borderRightStyle="solid"
      borderRightWidth={["borderWidth0", "borderWidth0", "borderWidth10"]}
      minWidth={subtitle ? "sizeSidebar" : "size0"}
    >
      <MediaObject verticalAlign="center">
        <MediaFigure spacing="space40">
          <LogoTwilioIcon
            decorative
            color="colorTextInverse"
            size="sizeIcon60"
          />
        </MediaFigure>
        <MediaBody>
          <Text
            as="div"
            fontSize="fontSize40"
            lineHeight="lineHeight40"
            color="colorTextInverse"
          >
            <StyledLink to="/">{title}</StyledLink>
          </Text>
          {subtitle ? (
            <Stack orientation="horizontal" spacing="space20">
              <Text
                as="span"
                fontSize="fontSize20"
                fontWeight="fontWeightNormal"
                lineHeight="lineHeight20"
                color="colorTextInverse"
              >
                {subtitle}
              </Text>
            </Stack>
          ) : null}
        </MediaBody>
      </MediaObject>
    </Box>
  );
};

export { SiteHeaderLogo };
