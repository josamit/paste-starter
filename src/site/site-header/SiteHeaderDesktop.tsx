import * as React from "react";
import { Box } from "@twilio-paste/box";
import { Stack } from "@twilio-paste/stack";
import { SiteHeaderSearch } from "./SiteHeaderSearch";
import { SiteHeaderLogo } from "./SiteHeaderLogo";
import { DarkModeToggle } from "./DarkModeToggle";

const SiteHeaderDesktop: React.VFC = () => {
  return (
    <Box
      as="header"
      display={["none", "none", "flex"]}
      alignItems="stretch"
      backgroundColor="colorBackgroundBodyInverse"
      borderBottomColor="colorBorderInverseWeaker"
      borderBottomWidth="borderWidth10"
      borderBottomStyle="solid"
      flex="1 0 auto"
    >
      <SiteHeaderLogo title="Paste Starter" subtitle="Admin template" />
      <Box
        display="flex"
        justifyContent="space-between"
        alignItems="center"
        flexDirection="row"
        paddingTop="space60"
        paddingRight="space70"
        paddingBottom="space60"
        paddingLeft="space70"
        width="100%"
      >
        <DarkModeToggle />
        <Box marginTop="space10">
          <Stack orientation="horizontal" spacing="space60">
            <SiteHeaderSearch />
          </Stack>
        </Box>
      </Box>
    </Box>
  );
};

export { SiteHeaderDesktop };
