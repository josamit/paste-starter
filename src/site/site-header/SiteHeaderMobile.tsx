import * as React from "react";
import { Box } from "@twilio-paste/box";
import { Button } from "@twilio-paste/button";
import { SearchIcon } from "@twilio-paste/icons/esm/SearchIcon";
import { HamburgerToggle } from "./HamburgerToggle";
import { SiteHeaderLogo } from "./SiteHeaderLogo";
import { MobileNavigation } from "./SiteHeaderMobileNavigation";
import { MobileSearch } from "./SiteHeaderMobileSearch";

const SiteHeaderMobile: React.VFC = () => {
  const [navigationOpen, setNavigationOpen] = React.useState(false);
  const [searchOpen, setSearchOpen] = React.useState(false);

  return (
    <Box
      as="header"
      display={["flex", "flex", "none"]}
      alignItems="center"
      justifyContent="space-between"
      backgroundColor="colorBackgroundBodyInverse"
      position="sticky"
      top="0px"
      zIndex="zIndex80"
    >
      <Box margin="space60">
        <Button
          variant="inverse"
          size="icon_small"
          onClick={() => setSearchOpen(!searchOpen)}
        >
          <Box paddingX="space20">
            <SearchIcon
              size="sizeIcon50"
              decorative={false}
              title="Search by Account SID (ACxxx)"
            />
          </Box>
        </Button>
        <MobileSearch
          isOpen={searchOpen}
          onClose={() => setSearchOpen(false)}
        />
      </Box>
      <SiteHeaderLogo title="Paste Starter" />
      <Box margin="space60">
        <Button
          data-testid="open_main_navigation"
          aria-label="Open main navigation"
          variant="inverse"
          size="icon_small"
          onClick={() => setNavigationOpen(!navigationOpen)}
        >
          <HamburgerToggle
            aria-hidden
            toggled={navigationOpen}
            color={
              navigationOpen
                ? "colorBackgroundBodyInverse"
                : "colorBackgroundBody"
            }
          />
        </Button>
        <MobileNavigation
          isOpen={navigationOpen}
          onClose={() => setNavigationOpen(false)}
        />
      </Box>
    </Box>
  );
};

export { SiteHeaderMobile };
