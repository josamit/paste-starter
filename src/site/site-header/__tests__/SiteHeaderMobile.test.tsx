import * as React from "react";
import { renderComponent } from "@src/utils/test-utils";
import { SiteHeaderMobile } from "@site/site-header/SiteHeaderMobile";
import { act, fireEvent } from "@testing-library/react";

describe("<SiteHeaderMobile />", () => {
  it("should render correctly", async () => {
    const { container } = renderComponent(<SiteHeaderMobile />);

    expect(container).toMatchSnapshot();
  });

  it("should be able to open and close main mobile navigation", async () => {
    const { container, getByTestId } = renderComponent(<SiteHeaderMobile />);

    await act(() => {
      fireEvent.click(getByTestId("open_main_navigation"));
    });

    expect(container).toMatchSnapshot("mobile main nav open");

    await act(() => {
      fireEvent.click(getByTestId("close_main_navigation"));
    });

    expect(container).toMatchSnapshot("mobile main nav close");
  });

  it("should be able to open and close mobile navigation", async () => {
    const { container, getByText } = renderComponent(<SiteHeaderMobile />);

    await act(() => {
      fireEvent.click(getByText("Search by Account SID (ACxxx)"));
    });

    expect(container).toMatchSnapshot("mobile nav open");

    await act(() => {
      fireEvent.click(getByText("Close Search"));
    });

    expect(container).toMatchSnapshot("mobile nav closed");
  });
});
