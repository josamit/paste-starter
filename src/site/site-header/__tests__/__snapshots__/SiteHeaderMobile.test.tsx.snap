// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SiteHeaderMobile /> should be able to open and close main mobile navigation: mobile main nav close 1`] = `
<div
  aria-hidden="true"
>
  <div
    class="css-ql39e"
  >
    <header
      class="css-15c1jec"
      data-paste-element="BOX"
      display="flex,flex,none"
    >
      <div
        class="css-c8ym1p"
        data-paste-element="BOX"
      >
        <button
          aria-busy="false"
          class="css-12ggu7m"
          color="colorTextInverse"
          cursor="pointer"
          data-paste-element="BUTTON"
          display="inline-block"
          font-family="fontFamilyText"
          font-weight="fontWeightSemibold"
          text-decoration="none"
          type="button"
          width="auto"
        >
          <span
            aria-hidden="false"
            class="css-cfzbzn"
            data-paste-element="BOX"
            display="flex"
            opacity="1"
            text-decoration="inherit"
          >
            <div
              class="css-j5k42x"
              data-paste-element="BOX"
            >
              <span
                class="css-zhc3mm"
                color="currentColor"
                data-paste-element="ICON"
                display="block"
              >
                <svg
                  aria-hidden="false"
                  height="100%"
                  role="img"
                  viewBox="0 0 20 20"
                  width="100%"
                >
                  <title>
                    Search by Account SID (ACxxx)
                  </title>
                  <path
                    d="M5.43 5.43a4.882 4.882 0 017.383 6.347l2.973 2.973a.732.732 0 01-1.036 1.036l-2.973-2.973A4.883 4.883 0 015.43 5.43zm1.035 1.035a3.417 3.417 0 104.833 4.833 3.417 3.417 0 00-4.833-4.833z"
                    fill="currentColor"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
            </div>
          </span>
        </button>
      </div>
      <div
        class="css-1pbe5qc"
        data-paste-element="BOX"
        display="flex"
      >
        <span
          class="css-8jah14"
          data-paste-element="MEDIA_OBJECT"
          display="flex"
        >
          <span
            class="css-and4r"
            data-paste-element="MEDIA_FIGURE"
            display="flex"
          >
            <span
              class="css-1v1gjpp"
              color="colorTextInverse"
              data-paste-element="ICON"
              display="block"
            >
              <svg
                aria-hidden="true"
                height="100%"
                role="img"
                viewBox="0 0 20 20"
                width="100%"
              >
                <path
                  d="M10 0C4.477 0 0 4.477 0 10s4.477 10 10 10 10-4.477 10-10A10 10 0 0010 0zm0 17.333a7.333 7.333 0 110-14.666 7.333 7.333 0 010 14.666zm4.533-9.8a2.067 2.067 0 11-2.066-2.066 2.08 2.08 0 012.066 2.066zm0 4.934a2.067 2.067 0 11-2.066-2.067 2.08 2.08 0 012.066 2.067zm-4.933 0A2.067 2.067 0 117.533 10.4 2.08 2.08 0 019.6 12.467zm0-4.934a2.067 2.067 0 11-2.067-2.066A2.08 2.08 0 019.6 7.533z"
                  fill="currentColor"
                  fill-rule="evenodd"
                />
              </svg>
            </span>
          </span>
          <span
            class="css-1i39mnm"
            data-paste-element="MEDIA_BODY"
          >
            <div
              class="css-6kf143"
              color="colorTextInverse"
              data-paste-element="TEXT"
              font-size="fontSize40"
            >
              <a
                class="css-mtfyde"
                href="/"
              >
                Paste Starter
              </a>
            </div>
          </span>
        </span>
      </div>
      <div
        class="css-c8ym1p"
        data-paste-element="BOX"
      >
        <button
          aria-busy="false"
          aria-label="Open main navigation"
          class="css-12ggu7m"
          color="colorTextInverse"
          cursor="pointer"
          data-paste-element="BUTTON"
          data-testid="open_main_navigation"
          display="inline-block"
          font-family="fontFamilyText"
          font-weight="fontWeightSemibold"
          text-decoration="none"
          type="button"
          width="auto"
        >
          <span
            aria-hidden="false"
            class="css-cfzbzn"
            data-paste-element="BOX"
            display="flex"
            opacity="1"
            text-decoration="inherit"
          >
            <div
              aria-hidden="true"
              class="css-dujrh8"
              data-paste-element="BOX"
              height="28px"
              width="36px"
            >
              <span
                class="css-yufnr1"
                data-paste-element="BOX"
                height="3px"
                opacity="1"
                transform="rotate(0deg)"
                width="16px"
              />
              <span
                class="css-rz76f9"
                data-paste-element="BOX"
                height="3px"
                transform="translateX(0) translateY(0) rotate(0deg)"
                width="16px"
              />
              <span
                class="css-rz76f9"
                data-paste-element="BOX"
                height="3px"
                transform="translateX(0) translateY(0) rotate(0deg)"
                width="16px"
              />
              <span
                class="css-1k16ztv"
                data-paste-element="BOX"
                height="3px"
                opacity="1"
                transform="translateX(0) translateY(0) rotate(0deg)"
                width="16px"
              />
            </div>
          </span>
        </button>
      </div>
    </header>
  </div>
</div>
`;

exports[`<SiteHeaderMobile /> should be able to open and close main mobile navigation: mobile main nav open 1`] = `
<div
  aria-hidden="true"
>
  <div
    class="css-ql39e"
  >
    <header
      class="css-15c1jec"
      data-paste-element="BOX"
      display="flex,flex,none"
    >
      <div
        class="css-c8ym1p"
        data-paste-element="BOX"
      >
        <button
          aria-busy="false"
          class="css-12ggu7m"
          color="colorTextInverse"
          cursor="pointer"
          data-paste-element="BUTTON"
          display="inline-block"
          font-family="fontFamilyText"
          font-weight="fontWeightSemibold"
          text-decoration="none"
          type="button"
          width="auto"
        >
          <span
            aria-hidden="false"
            class="css-cfzbzn"
            data-paste-element="BOX"
            display="flex"
            opacity="1"
            text-decoration="inherit"
          >
            <div
              class="css-j5k42x"
              data-paste-element="BOX"
            >
              <span
                class="css-zhc3mm"
                color="currentColor"
                data-paste-element="ICON"
                display="block"
              >
                <svg
                  aria-hidden="false"
                  height="100%"
                  role="img"
                  viewBox="0 0 20 20"
                  width="100%"
                >
                  <title>
                    Search by Account SID (ACxxx)
                  </title>
                  <path
                    d="M5.43 5.43a4.882 4.882 0 017.383 6.347l2.973 2.973a.732.732 0 01-1.036 1.036l-2.973-2.973A4.883 4.883 0 015.43 5.43zm1.035 1.035a3.417 3.417 0 104.833 4.833 3.417 3.417 0 00-4.833-4.833z"
                    fill="currentColor"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
            </div>
          </span>
        </button>
      </div>
      <div
        class="css-1pbe5qc"
        data-paste-element="BOX"
        display="flex"
      >
        <span
          class="css-8jah14"
          data-paste-element="MEDIA_OBJECT"
          display="flex"
        >
          <span
            class="css-and4r"
            data-paste-element="MEDIA_FIGURE"
            display="flex"
          >
            <span
              class="css-1v1gjpp"
              color="colorTextInverse"
              data-paste-element="ICON"
              display="block"
            >
              <svg
                aria-hidden="true"
                height="100%"
                role="img"
                viewBox="0 0 20 20"
                width="100%"
              >
                <path
                  d="M10 0C4.477 0 0 4.477 0 10s4.477 10 10 10 10-4.477 10-10A10 10 0 0010 0zm0 17.333a7.333 7.333 0 110-14.666 7.333 7.333 0 010 14.666zm4.533-9.8a2.067 2.067 0 11-2.066-2.066 2.08 2.08 0 012.066 2.066zm0 4.934a2.067 2.067 0 11-2.066-2.067 2.08 2.08 0 012.066 2.067zm-4.933 0A2.067 2.067 0 117.533 10.4 2.08 2.08 0 019.6 12.467zm0-4.934a2.067 2.067 0 11-2.067-2.066A2.08 2.08 0 019.6 7.533z"
                  fill="currentColor"
                  fill-rule="evenodd"
                />
              </svg>
            </span>
          </span>
          <span
            class="css-1i39mnm"
            data-paste-element="MEDIA_BODY"
          >
            <div
              class="css-6kf143"
              color="colorTextInverse"
              data-paste-element="TEXT"
              font-size="fontSize40"
            >
              <a
                class="css-mtfyde"
                href="/"
              >
                Paste Starter
              </a>
            </div>
          </span>
        </span>
      </div>
      <div
        class="css-c8ym1p"
        data-paste-element="BOX"
      >
        <button
          aria-busy="false"
          aria-label="Open main navigation"
          class="css-12ggu7m"
          color="colorTextInverse"
          cursor="pointer"
          data-paste-element="BUTTON"
          data-testid="open_main_navigation"
          display="inline-block"
          font-family="fontFamilyText"
          font-weight="fontWeightSemibold"
          text-decoration="none"
          type="button"
          width="auto"
        >
          <span
            aria-hidden="false"
            class="css-cfzbzn"
            data-paste-element="BOX"
            display="flex"
            opacity="1"
            text-decoration="inherit"
          >
            <div
              aria-hidden="true"
              class="css-dujrh8"
              data-paste-element="BOX"
              height="28px"
              width="36px"
            >
              <span
                class="css-15ywz3f"
                data-paste-element="BOX"
                height="3px"
                opacity="0"
                transform="translateX(6px) translateY(4px) rotate(45deg)"
                width="16px"
              />
              <span
                class="css-25g0e5"
                data-paste-element="BOX"
                height="3px"
                transform="translateX(2px) translateY(-5px) rotate(45deg)"
                width="16px"
              />
              <span
                class="css-1j1o2c5"
                data-paste-element="BOX"
                height="3px"
                transform="translateX(2px) translateY(6px) rotate(-45deg)"
                width="16px"
              />
              <span
                class="css-d7i4zw"
                data-paste-element="BOX"
                height="3px"
                opacity="0"
                transform="translateX(6px) translateY(-4px) rotate(-45deg)"
                width="16px"
              />
            </div>
          </span>
        </button>
      </div>
    </header>
  </div>
</div>
`;

exports[`<SiteHeaderMobile /> should be able to open and close mobile navigation: mobile nav closed 1`] = `
<div>
  <div
    class="css-ql39e"
  >
    <header
      class="css-15c1jec"
      data-paste-element="BOX"
      display="flex,flex,none"
    >
      <div
        class="css-c8ym1p"
        data-paste-element="BOX"
      >
        <button
          aria-busy="false"
          class="css-12ggu7m"
          color="colorTextInverse"
          cursor="pointer"
          data-paste-element="BUTTON"
          display="inline-block"
          font-family="fontFamilyText"
          font-weight="fontWeightSemibold"
          text-decoration="none"
          type="button"
          width="auto"
        >
          <span
            aria-hidden="false"
            class="css-cfzbzn"
            data-paste-element="BOX"
            display="flex"
            opacity="1"
            text-decoration="inherit"
          >
            <div
              class="css-j5k42x"
              data-paste-element="BOX"
            >
              <span
                class="css-zhc3mm"
                color="currentColor"
                data-paste-element="ICON"
                display="block"
              >
                <svg
                  aria-hidden="false"
                  height="100%"
                  role="img"
                  viewBox="0 0 20 20"
                  width="100%"
                >
                  <title>
                    Search by Account SID (ACxxx)
                  </title>
                  <path
                    d="M5.43 5.43a4.882 4.882 0 017.383 6.347l2.973 2.973a.732.732 0 01-1.036 1.036l-2.973-2.973A4.883 4.883 0 015.43 5.43zm1.035 1.035a3.417 3.417 0 104.833 4.833 3.417 3.417 0 00-4.833-4.833z"
                    fill="currentColor"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
            </div>
          </span>
        </button>
      </div>
      <div
        class="css-1pbe5qc"
        data-paste-element="BOX"
        display="flex"
      >
        <span
          class="css-8jah14"
          data-paste-element="MEDIA_OBJECT"
          display="flex"
        >
          <span
            class="css-and4r"
            data-paste-element="MEDIA_FIGURE"
            display="flex"
          >
            <span
              class="css-1v1gjpp"
              color="colorTextInverse"
              data-paste-element="ICON"
              display="block"
            >
              <svg
                aria-hidden="true"
                height="100%"
                role="img"
                viewBox="0 0 20 20"
                width="100%"
              >
                <path
                  d="M10 0C4.477 0 0 4.477 0 10s4.477 10 10 10 10-4.477 10-10A10 10 0 0010 0zm0 17.333a7.333 7.333 0 110-14.666 7.333 7.333 0 010 14.666zm4.533-9.8a2.067 2.067 0 11-2.066-2.066 2.08 2.08 0 012.066 2.066zm0 4.934a2.067 2.067 0 11-2.066-2.067 2.08 2.08 0 012.066 2.067zm-4.933 0A2.067 2.067 0 117.533 10.4 2.08 2.08 0 019.6 12.467zm0-4.934a2.067 2.067 0 11-2.067-2.066A2.08 2.08 0 019.6 7.533z"
                  fill="currentColor"
                  fill-rule="evenodd"
                />
              </svg>
            </span>
          </span>
          <span
            class="css-1i39mnm"
            data-paste-element="MEDIA_BODY"
          >
            <div
              class="css-6kf143"
              color="colorTextInverse"
              data-paste-element="TEXT"
              font-size="fontSize40"
            >
              <a
                class="css-mtfyde"
                href="/"
              >
                Paste Starter
              </a>
            </div>
          </span>
        </span>
      </div>
      <div
        class="css-c8ym1p"
        data-paste-element="BOX"
      >
        <button
          aria-busy="false"
          aria-label="Open main navigation"
          class="css-12ggu7m"
          color="colorTextInverse"
          cursor="pointer"
          data-paste-element="BUTTON"
          data-testid="open_main_navigation"
          display="inline-block"
          font-family="fontFamilyText"
          font-weight="fontWeightSemibold"
          text-decoration="none"
          type="button"
          width="auto"
        >
          <span
            aria-hidden="false"
            class="css-cfzbzn"
            data-paste-element="BOX"
            display="flex"
            opacity="1"
            text-decoration="inherit"
          >
            <div
              aria-hidden="true"
              class="css-dujrh8"
              data-paste-element="BOX"
              height="28px"
              width="36px"
            >
              <span
                class="css-yufnr1"
                data-paste-element="BOX"
                height="3px"
                opacity="1"
                transform="rotate(0deg)"
                width="16px"
              />
              <span
                class="css-rz76f9"
                data-paste-element="BOX"
                height="3px"
                transform="translateX(0) translateY(0) rotate(0deg)"
                width="16px"
              />
              <span
                class="css-rz76f9"
                data-paste-element="BOX"
                height="3px"
                transform="translateX(0) translateY(0) rotate(0deg)"
                width="16px"
              />
              <span
                class="css-1k16ztv"
                data-paste-element="BOX"
                height="3px"
                opacity="1"
                transform="translateX(0) translateY(0) rotate(0deg)"
                width="16px"
              />
            </div>
          </span>
        </button>
      </div>
    </header>
  </div>
</div>
`;

exports[`<SiteHeaderMobile /> should be able to open and close mobile navigation: mobile nav open 1`] = `
<div
  aria-hidden="true"
>
  <div
    class="css-ql39e"
  >
    <header
      class="css-15c1jec"
      data-paste-element="BOX"
      display="flex,flex,none"
    >
      <div
        class="css-c8ym1p"
        data-paste-element="BOX"
      >
        <button
          aria-busy="false"
          class="css-12ggu7m"
          color="colorTextInverse"
          cursor="pointer"
          data-paste-element="BUTTON"
          display="inline-block"
          font-family="fontFamilyText"
          font-weight="fontWeightSemibold"
          text-decoration="none"
          type="button"
          width="auto"
        >
          <span
            aria-hidden="false"
            class="css-cfzbzn"
            data-paste-element="BOX"
            display="flex"
            opacity="1"
            text-decoration="inherit"
          >
            <div
              class="css-j5k42x"
              data-paste-element="BOX"
            >
              <span
                class="css-zhc3mm"
                color="currentColor"
                data-paste-element="ICON"
                display="block"
              >
                <svg
                  aria-hidden="false"
                  height="100%"
                  role="img"
                  viewBox="0 0 20 20"
                  width="100%"
                >
                  <title>
                    Search by Account SID (ACxxx)
                  </title>
                  <path
                    d="M5.43 5.43a4.882 4.882 0 017.383 6.347l2.973 2.973a.732.732 0 01-1.036 1.036l-2.973-2.973A4.883 4.883 0 015.43 5.43zm1.035 1.035a3.417 3.417 0 104.833 4.833 3.417 3.417 0 00-4.833-4.833z"
                    fill="currentColor"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
            </div>
          </span>
        </button>
      </div>
      <div
        class="css-1pbe5qc"
        data-paste-element="BOX"
        display="flex"
      >
        <span
          class="css-8jah14"
          data-paste-element="MEDIA_OBJECT"
          display="flex"
        >
          <span
            class="css-and4r"
            data-paste-element="MEDIA_FIGURE"
            display="flex"
          >
            <span
              class="css-1v1gjpp"
              color="colorTextInverse"
              data-paste-element="ICON"
              display="block"
            >
              <svg
                aria-hidden="true"
                height="100%"
                role="img"
                viewBox="0 0 20 20"
                width="100%"
              >
                <path
                  d="M10 0C4.477 0 0 4.477 0 10s4.477 10 10 10 10-4.477 10-10A10 10 0 0010 0zm0 17.333a7.333 7.333 0 110-14.666 7.333 7.333 0 010 14.666zm4.533-9.8a2.067 2.067 0 11-2.066-2.066 2.08 2.08 0 012.066 2.066zm0 4.934a2.067 2.067 0 11-2.066-2.067 2.08 2.08 0 012.066 2.067zm-4.933 0A2.067 2.067 0 117.533 10.4 2.08 2.08 0 019.6 12.467zm0-4.934a2.067 2.067 0 11-2.067-2.066A2.08 2.08 0 019.6 7.533z"
                  fill="currentColor"
                  fill-rule="evenodd"
                />
              </svg>
            </span>
          </span>
          <span
            class="css-1i39mnm"
            data-paste-element="MEDIA_BODY"
          >
            <div
              class="css-6kf143"
              color="colorTextInverse"
              data-paste-element="TEXT"
              font-size="fontSize40"
            >
              <a
                class="css-mtfyde"
                href="/"
              >
                Paste Starter
              </a>
            </div>
          </span>
        </span>
      </div>
      <div
        class="css-c8ym1p"
        data-paste-element="BOX"
      >
        <button
          aria-busy="false"
          aria-label="Open main navigation"
          class="css-12ggu7m"
          color="colorTextInverse"
          cursor="pointer"
          data-paste-element="BUTTON"
          data-testid="open_main_navigation"
          display="inline-block"
          font-family="fontFamilyText"
          font-weight="fontWeightSemibold"
          text-decoration="none"
          type="button"
          width="auto"
        >
          <span
            aria-hidden="false"
            class="css-cfzbzn"
            data-paste-element="BOX"
            display="flex"
            opacity="1"
            text-decoration="inherit"
          >
            <div
              aria-hidden="true"
              class="css-dujrh8"
              data-paste-element="BOX"
              height="28px"
              width="36px"
            >
              <span
                class="css-yufnr1"
                data-paste-element="BOX"
                height="3px"
                opacity="1"
                transform="rotate(0deg)"
                width="16px"
              />
              <span
                class="css-rz76f9"
                data-paste-element="BOX"
                height="3px"
                transform="translateX(0) translateY(0) rotate(0deg)"
                width="16px"
              />
              <span
                class="css-rz76f9"
                data-paste-element="BOX"
                height="3px"
                transform="translateX(0) translateY(0) rotate(0deg)"
                width="16px"
              />
              <span
                class="css-1k16ztv"
                data-paste-element="BOX"
                height="3px"
                opacity="1"
                transform="translateX(0) translateY(0) rotate(0deg)"
                width="16px"
              />
            </div>
          </span>
        </button>
      </div>
    </header>
  </div>
</div>
`;

exports[`<SiteHeaderMobile /> should render correctly 1`] = `
<div>
  <div
    class="css-ql39e"
  >
    <header
      class="css-15c1jec"
      data-paste-element="BOX"
      display="flex,flex,none"
    >
      <div
        class="css-c8ym1p"
        data-paste-element="BOX"
      >
        <button
          aria-busy="false"
          class="css-12ggu7m"
          color="colorTextInverse"
          cursor="pointer"
          data-paste-element="BUTTON"
          display="inline-block"
          font-family="fontFamilyText"
          font-weight="fontWeightSemibold"
          text-decoration="none"
          type="button"
          width="auto"
        >
          <span
            aria-hidden="false"
            class="css-cfzbzn"
            data-paste-element="BOX"
            display="flex"
            opacity="1"
            text-decoration="inherit"
          >
            <div
              class="css-j5k42x"
              data-paste-element="BOX"
            >
              <span
                class="css-zhc3mm"
                color="currentColor"
                data-paste-element="ICON"
                display="block"
              >
                <svg
                  aria-hidden="false"
                  height="100%"
                  role="img"
                  viewBox="0 0 20 20"
                  width="100%"
                >
                  <title>
                    Search by Account SID (ACxxx)
                  </title>
                  <path
                    d="M5.43 5.43a4.882 4.882 0 017.383 6.347l2.973 2.973a.732.732 0 01-1.036 1.036l-2.973-2.973A4.883 4.883 0 015.43 5.43zm1.035 1.035a3.417 3.417 0 104.833 4.833 3.417 3.417 0 00-4.833-4.833z"
                    fill="currentColor"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
            </div>
          </span>
        </button>
      </div>
      <div
        class="css-1pbe5qc"
        data-paste-element="BOX"
        display="flex"
      >
        <span
          class="css-8jah14"
          data-paste-element="MEDIA_OBJECT"
          display="flex"
        >
          <span
            class="css-and4r"
            data-paste-element="MEDIA_FIGURE"
            display="flex"
          >
            <span
              class="css-1v1gjpp"
              color="colorTextInverse"
              data-paste-element="ICON"
              display="block"
            >
              <svg
                aria-hidden="true"
                height="100%"
                role="img"
                viewBox="0 0 20 20"
                width="100%"
              >
                <path
                  d="M10 0C4.477 0 0 4.477 0 10s4.477 10 10 10 10-4.477 10-10A10 10 0 0010 0zm0 17.333a7.333 7.333 0 110-14.666 7.333 7.333 0 010 14.666zm4.533-9.8a2.067 2.067 0 11-2.066-2.066 2.08 2.08 0 012.066 2.066zm0 4.934a2.067 2.067 0 11-2.066-2.067 2.08 2.08 0 012.066 2.067zm-4.933 0A2.067 2.067 0 117.533 10.4 2.08 2.08 0 019.6 12.467zm0-4.934a2.067 2.067 0 11-2.067-2.066A2.08 2.08 0 019.6 7.533z"
                  fill="currentColor"
                  fill-rule="evenodd"
                />
              </svg>
            </span>
          </span>
          <span
            class="css-1i39mnm"
            data-paste-element="MEDIA_BODY"
          >
            <div
              class="css-6kf143"
              color="colorTextInverse"
              data-paste-element="TEXT"
              font-size="fontSize40"
            >
              <a
                class="css-mtfyde"
                href="/"
              >
                Paste Starter
              </a>
            </div>
          </span>
        </span>
      </div>
      <div
        class="css-c8ym1p"
        data-paste-element="BOX"
      >
        <button
          aria-busy="false"
          aria-label="Open main navigation"
          class="css-12ggu7m"
          color="colorTextInverse"
          cursor="pointer"
          data-paste-element="BUTTON"
          data-testid="open_main_navigation"
          display="inline-block"
          font-family="fontFamilyText"
          font-weight="fontWeightSemibold"
          text-decoration="none"
          type="button"
          width="auto"
        >
          <span
            aria-hidden="false"
            class="css-cfzbzn"
            data-paste-element="BOX"
            display="flex"
            opacity="1"
            text-decoration="inherit"
          >
            <div
              aria-hidden="true"
              class="css-dujrh8"
              data-paste-element="BOX"
              height="28px"
              width="36px"
            >
              <span
                class="css-yufnr1"
                data-paste-element="BOX"
                height="3px"
                opacity="1"
                transform="rotate(0deg)"
                width="16px"
              />
              <span
                class="css-rz76f9"
                data-paste-element="BOX"
                height="3px"
                transform="translateX(0) translateY(0) rotate(0deg)"
                width="16px"
              />
              <span
                class="css-rz76f9"
                data-paste-element="BOX"
                height="3px"
                transform="translateX(0) translateY(0) rotate(0deg)"
                width="16px"
              />
              <span
                class="css-1k16ztv"
                data-paste-element="BOX"
                height="3px"
                opacity="1"
                transform="translateX(0) translateY(0) rotate(0deg)"
                width="16px"
              />
            </div>
          </span>
        </button>
      </div>
    </header>
  </div>
</div>
`;
